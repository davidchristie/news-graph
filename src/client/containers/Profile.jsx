import React, { PropTypes } from 'react'
import { connect } from 'react-redux'

import { fetchProfile } from '../actions/profiles'
import Jumbotron from '../components/Jumbotron'
import Posts from '../components/Posts'
import PostArticle from './PostArticle'

export const Profile = class Profile extends React.Component {
  componentDidMount () {
    this.props.fetchProfile()
  }
  render () {
    const profile = this.props.profile
    const { name, posts, token } = profile
    return (
      <div>
        <Jumbotron title={name} />
        <div className='card'>
          <div className='card-block'>
            {token ? <PostArticle profile={profile} /> : null}
          </div>
        </div>
        {posts ? <Posts posts={posts} /> : null}
      </div>
    )
  }
}

Profile.propTypes = {
  fetchProfile: PropTypes.func.isRequired,
  profile: PropTypes.shape({
    name: PropTypes.string,
    posts: PropTypes.arrayOf(PropTypes.object),
    token: PropTypes.string
  }).isRequired
}

const mapStateToProps = (state, ownProps) => {
  const id = Number(ownProps.params.id)
  const profile = Object.assign({}, state.profiles[id])
  const account = state.account
  if (account.profile && account.profile.id === profile.id) {
    profile.token = account.profile.token
  }
  return {profile}
}

const mapDispatchToProps = (dispatch, ownProps) => {
  return {
    fetchProfile: () => {
      const userId = Number(ownProps.params.id)
      dispatch(fetchProfile({userId}))
    }
  }
}

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(Profile)
