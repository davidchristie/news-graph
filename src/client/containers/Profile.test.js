/* eslint-env jest */

import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'

import { Profile } from './Profile'

const params = {id: '1'}

test('Profile renders correctly', () => {
  const wrapper = shallow(<Profile params={params} />)
  expect(toJson(wrapper)).toMatchSnapshot()
})
