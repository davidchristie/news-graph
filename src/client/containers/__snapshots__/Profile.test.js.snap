exports[`test Profile renders correctly 1`] = `
<div>
  <Jumbotron
    title="Profile 1" />
  <div
    className="card">
    <div
      className="card-block">
      <div>
        <h2>
          Public
        </h2>
      </div>
    </div>
  </div>
  <Connect(Posts)
    userId={1} />
</div>
`;
